web/controllers/stream_controller.ex

```diff
@@ -50,6 +50,39 @@ defmodule Streamr.StreamController do
     render conn, "show.json-api", data: stream
   end
 
+  def delete(conn, %{"id" => id}) do
+    stream = Repo.get!(Stream, id)
+
+    case Repo.delete(stream) do
+    {:ok, stream} ->
+      conn
+      |> put_status(204)
+      |> render("show.json-api", data: Repo.preload(stream, :user))
```

Comment:
Can you change this rendering call to `send_resp(conn, 204, "")`? We don't really need to serialize the stream if it's being deleted.